src/leo_s3_auth.erl

```diff
@@ -34,20 +34,26 @@
 -include("leo_s3_user.hrl").
 -include_lib("eunit/include/eunit.hrl").
 -include_lib("stdlib/include/qlc.hrl").
+-include_lib("leo_logger/include/leo_logger.hrl").
 
 -export([start/2,
          create_table/2, put/1, bulk_put/1,
          update_providers/1,
          create_key/1, get_credential/1, has_credential/1, has_credential/2,
-         authenticate/3, get_signature/2,
+         authenticate/3, get_signature/3,
          find_all/0, checksum/0
         ]).
 
 
+-record(sign_v4_params, {credential     :: binary(),
+                         signature      :: binary(),
+                         signed_headers :: binary()
```

Comment:
>> ? These parameters look binaries to me.

yep. 
you are right.

>> BTW, I will check how to use dialyzer

At first,
issue the below command only once in the repo root. ( basically when you cloned into local )
```
make build_plt
```

issue the below command if you want to dialyze. ( basically when you finished to implement before UT )
```
make dialyzer
```